bleak/backends/bluezdbus/manager.py

```diff
@@ -804,21 +807,21 @@ async def _wait_condition(
 
         event = asyncio.Event()
 
-        def callback():
-            if (
-                self._properties[device_path][defs.DEVICE_INTERFACE][property_name]
-                == property_value
-            ):
+        def _wait_condition_callback(changed: Dict[str, Any]) -> None:
+            """Callback for when a property changes."""
+            if changed.get(property_name) == property_value:
                 event.set()
 
         device_callbacks = self._condition_callbacks.setdefault(device_path, set())
-        device_callbacks.add(callback)
+        device_callbacks.add(_wait_condition_callback)
 
         try:
             # can be canceled
             await event.wait()
         finally:
-            device_callbacks.remove(callback)
+            device_callbacks.remove(_wait_condition_callback)
+            if not device_callbacks:
+                del self._condition_callbacks[device_path]
```

Comment:
Same here, seems simpler to just leave the set.

Reply:
https://github.com/hbldh/bleak/pull/1400/files#r1307604639